server/operations/src/main/java/org/kaaproject/kaa/server/operations/service/akka/actors/core/user/GlobalUserActorMessageProcessor.java

```diff
@@ -0,0 +1,264 @@
+                    return false;
+                }
+            } else if (!appToken.equals(other.appToken)) {
+                return false;
+            }
+            if (schemaVersion != other.schemaVersion) {
+                return false;
+            }
+            return true;
+        }
+
+        @Override
+        public String toString() {
+            StringBuilder builder = new StringBuilder();
+            builder.append("ConfigurationKey [schemaVersion=");
+            builder.append(schemaVersion);
+            builder.append(", appToken=");
+            builder.append(appToken);
+            builder.append("]");
+            return builder.toString();
+        }
+    }
+
+    private static class GlobalRouteTable<T> {
+
+        private final Map<T, Set<GlobalRouteInfo>> routes;
+
+        public GlobalRouteTable() {
+            routes = new HashMap<>();
+        }
+
+        public boolean add(T key, GlobalRouteInfo route) {
+            Set<GlobalRouteInfo> keyRoutes = routes.get(key);
+            if (keyRoutes == null) {
+                keyRoutes = new HashSet<>();
+                routes.put(key, keyRoutes);
+            }
+            return keyRoutes.add(route);
+        }
+
+        public boolean remove(GlobalRouteInfo route) {
+            boolean found = false;
+            for(T key : routes.keySet()){
+                found = found || routes.get(key).remove(route);
+            }
+            return found;
+        }
+
+        public Set<GlobalRouteInfo> getRoutes(T key) {
+            return notNull(routes.get(key));
+        }
+
+        public Set<GlobalRouteInfo> getLocalRoutes(T key) {
+            Set<GlobalRouteInfo> result = new HashSet<GlobalRouteInfo>();
+            for (GlobalRouteInfo route : getRoutes(key)) {
+                String serverId = route.getAddress().getServerId();
+                if (serverId != null) {
+                    continue;
+                }
+                result.add(route);
+            }
+            return notNull(result);
+        }
+
+        public Map<String, Set<GlobalRouteInfo>> getRemoteRoutes(T key) {
+            Map<String, Set<GlobalRouteInfo>> result = new HashMap<String, Set<GlobalRouteInfo>>();
+            for (GlobalRouteInfo route : getRoutes(key)) {
+                String serverId = route.getAddress().getServerId();
+                if (serverId == null) {
+                    continue;
+                }
+                Set<GlobalRouteInfo> set = result.get(serverId);
+                if (set == null) {
+                    set = new HashSet<GlobalRouteInfo>();
+                    result.put(serverId, set);
+                }
+                set.add(route);
+            }
+            return result;
+        }
+
+        public void clear() {
+            routes.clear();
+        }
+        
+        private static Set<GlobalRouteInfo> notNull(Set<GlobalRouteInfo> result) {
+            if(result != null){
```

Comment:
The prettier version with code style spaces:
if (result != null) {
    return result;
}
return Collections.emptySet();